File: ConversationIDE/src/VanEngine.Core/VAN/garden_config.py
from __future__ import annotations

from pathlib import Path


class GardenConfig:
    def __init__(self, state_root: str = "", schema_root: str = "",
                 registry_root: str = "", policy_root: str = ""):
        self.state_root = state_root
        self.schema_root = schema_root
        self.registry_root = registry_root
        self.policy_root = policy_root

    @property
    def is_complete(self) -> bool:
        return all([self.state_root, self.schema_root, self.registry_root, self.policy_root])

    @staticmethod
    def from_directory(dir_path: str) -> GardenConfig:
        d = Path(dir_path)
        return GardenConfig(
            state_root=str(d / "garden_one_state.json"),
            schema_root=str(d / "garden_two_schema.json"),
            registry_root=str(d / "garden_three_registry.json"),
            policy_root=str(d / "garden_four_network_policy.json"),
        )

    def all_paths(self) -> list[str]:
        return [self.state_root, self.schema_root, self.registry_root, self.policy_root]

File: engine/VanEngine.Core/VAN/GardenConfig.cs
namespace VanEngine.Core.VAN;

public sealed class GardenConfig
{
    public string StateRoot { get; set; } = string.Empty;
    public string SchemaRoot { get; set; } = string.Empty;
    public string RegistryRoot { get; set; } = string.Empty;
    public string PolicyRoot { get; set; } = string.Empty;

    public bool IsComplete =>
        !string.IsNullOrEmpty(StateRoot) &&
        !string.IsNullOrEmpty(SchemaRoot) &&
        !string.IsNullOrEmpty(RegistryRoot) &&
        !string.IsNullOrEmpty(PolicyRoot);

    public static GardenConfig FromDirectory(string dir)
    {
        return new GardenConfig
        {
            StateRoot = Path.Combine(dir, "garden_one_state.json"),
            SchemaRoot = Path.Combine(dir, "garden_two_schema.json"),
            RegistryRoot = Path.Combine(dir, "garden_three_registry.json"),
            PolicyRoot = Path.Combine(dir, "garden_four_network_policy.json")
        };
    }

    public string[] AllPaths() =>
        new[] { StateRoot, SchemaRoot, RegistryRoot, PolicyRoot };
}
